go/apps/api/openapi/spec/common/addPermissionReference.yaml

```
@@ -0,0 +1,40 @@
+      Enables automatic permission creation when the specified slug does not exist.
+      Only works with slug-based references, not ID-based references.
+      Requires the `rbac.*.create_permission` permission on your root key.
+
+      Created permissions are permanent and visible workspace-wide to all API keys.
+      Use carefully to avoid permission proliferation from typos or uncontrolled creation.
+      Consider centralizing permission creation in controlled processes for better governance.
+      Auto-created permissions use the slug as both the name and identifier.
+additionalProperties: false
+example:
+  slug: documents.write
+  create: true
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)